app/src/main/java/memphis/myapplication/SharedPrefsManager.java

```diff
@@ -0,0 +1,64 @@
+package memphis.myapplication;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+public class SharedPrefsManager {
+    private static final String KEY_USERNAME = "username";
+    private static final String KEY_PASSWORD = "password";
+    private static final String KEY_LOGIN_STATUS = "login_status";
+
+    private String mUsername;
+    private String mPassword;
+    private Boolean mLogInStatus;
+
+    private SharedPreferences mSharedPreferences;
+    private static SharedPrefsManager sharedPrefsManager;
+
+    public static SharedPrefsManager getInstance(Context context) {
+        if (sharedPrefsManager == null) {
+            sharedPrefsManager = new SharedPrefsManager(context);
+        }
+        return sharedPrefsManager;
+    }
+    private SharedPrefsManager(Context context){
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        mUsername = mSharedPreferences.getString(KEY_USERNAME, null);
+        mPassword = mSharedPreferences.getString(KEY_PASSWORD, null);
+        mLogInStatus = mSharedPreferences.getBoolean(KEY_LOGIN_STATUS, false);
+    }
+
+    public String getUsername() {
+        return mUsername;
+    }
+
+    public String getPassword() {
+        return mPassword;
+    }
+
+    public Boolean getLogInStatus() {
+        return mLogInStatus;
+    }
+
+    public void setCredentials(String username, String password) {
+        mLogInStatus = true;
+        mUsername = username;
+        mPassword = password;
+        SharedPreferences.Editor editor =  mSharedPreferences.edit();
+        editor.putString(KEY_USERNAME, username);
+        editor.putString(KEY_PASSWORD, password);
+        editor.apply();
+    }
+
+    public void removeCredentials(){
+        mLogInStatus = false;
+        mPassword = null;
+        mUsername = null;
+        SharedPreferences.Editor editor =  mSharedPreferences.edit();
+        editor.remove(KEY_USERNAME);
+        editor.remove(KEY_PASSWORD);
+        editor.apply();
+    }
+
+}
```

Comment:
I like the inclusion of removeCredentials. After the friends list and keys are added, it would be a good idea to make this something developers can click somewhere to reset the app.